test/common/FileEncoding/FileEncoding.Tests.ps1

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Describe 'Verify file encoding' {
+    BeforeAll {
+        Push-Location $PSScriptRoot\..\..\..
+        $nonBinaryFiles = (git grep -I --files-with-matches -e $'')
+    }
+
+    AfterAll {
+        Pop-Location
+    }
+
+    It 'No Unicode BOM' {
+        filter hasBom {
+            $_.Where{
+                $bytes = [byte[]]::new(3)
+                $stream = [System.IO.File]::OpenRead($_)
+                $stream.Read($bytes, 0, 3) > $null
+                $stream.Close()
+                $bytes[0] -eq 0xef -and $bytes[1] -eq 0xbb -and $bytes[2] -eq 0xbf
```

Comment:
There are some BOM kinds. I'd expect we check all.

Reply:
There is only one kind of BOM for UTF-8. Perhaps in new PR we could verify all Unicode files are in UTF-8, not UTF-16, UTF-32, etc.

Reply:
I don't understand why does not check the 3 bytes for BOMs in one place.

Reply:
The test should be "No BOM" and check a preambula for all cases.
https://www.powershellgallery.com/packages/PSTemplatizer/1.0.20/Content/Functions%5CGet-FileEncoding.ps1